### PR TITLE
fix(operator): settings key were transform twice

### PR DESCRIPTION
**Pull request by Dav-14**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new method for retrieving the Caddy image to improve consistency across components.
  - Enhanced image handling by normalizing versioning and setting up registry paths more efficiently.
- **Bug Fixes**
  - Adjusted settings retrieval logic to accommodate keys with dots, ensuring more accurate configuration parsing.
- **Refactor**
  - Updated various components to use nested keys instead of concatenated strings for settings, enhancing clarity and reliability in configuration management.
- **Tests**
  - Expanded testing to cover new settings parsing logic and key handling improvements.

### CODE REVIEW

File: components/operator/internal/resources/settings/helpers.go [hunk 1]
```diff
@@ -310,11 +310,7 @@ func findMatchingSettings(settings []v1beta1.Settings, keys ...string) (*string,
 	// Keys can be passed as "a.b.c", instead of "a", "b", "c"
 	// Keys can be passed as "a.b.*", instead of "a", "b", "*"
 	// Keys can be passed as "a.*.c", instead of "a", "*", "c"
-	// Keys can be passed as "a."*.*".c, instead of "a", "b", "c"
-	keys = Flatten(Map(keys, func(from string) []string {
```
Comment (Dav-14) [inline components/operator/internal/resources/settings/helpers.go:314]: Redondant flattent, findMatchingSettings need already flatten keys

File: components/operator/internal/resources/ledgers/deployments.go [hunk 2]
```diff
@@ -301,7 +302,7 @@ func createGatewayDeployment(ctx core.Context, stack *v1beta1.Stack, ledger *v1b
 	env = append(env, otlpEnv...)
 	env = append(env, core.GetDevEnvVars(stack, ledger)...)
 
-	caddyImage, err := settings.GetStringOrDefault(ctx, stack.Name, "caddy:2.7.6-alpine", "caddy.image")
+	caddyImage, err := registries.GetCaddyImage(ctx, stack, "2.7.6-alpine")
```
Comment (Dav-14) [inline components/operator/internal/resources/ledgers/deployments.go:305]: Migrate caddy image setting in new one

File: components/operator/internal/resources/payments/deployments.go [hunk 3]
```diff
@@ -236,7 +237,7 @@ func createGateway(ctx core.Context, stack *v1beta1.Stack, p *v1beta1.Payments)
 	env = append(env, otlpEnv...)
 	env = append(env, core.GetDevEnvVars(stack, p)...)
 
-	caddyImage, err := settings.GetStringOrDefault(ctx, stack.Name, "caddy:2.7.6-alpine", "caddy.image")
+	caddyImage, err := registries.GetCaddyImage(ctx, stack, "2.7.6-alpine")
```
Comment (Dav-14) [inline components/operator/internal/resources/payments/deployments.go:240]: same migration

File: components/operator/internal/resources/settings/settings.go [hunk 4]
```diff
@@ -18,7 +17,7 @@ import (
 func init() {
 	core.Init(
 		core.WithSimpleIndex[*v1beta1.Settings]("keylen", func(t *v1beta1.Settings) string {
-			return fmt.Sprint(len(strings.Split(t.Spec.Key, ".")))
+			return fmt.Sprint(len(splitKeywordWithDot(t.Spec.Key)))
```
Comment (Dav-14) [inline components/operator/internal/resources/settings/settings.go:20]: Index has to split key following the rule

File: components/operator/internal/resources/registries/registries.go [hunk 5]
```diff
@@ -9,37 +9,33 @@ import (
 	"github.com/formancehq/operator/internal/core"
 )
 
+// ghcr.io/<organization>/<repository>:<version>
+// public.ecr.aws/<organization>/jeffail/benthos:<version>
+// docker.io/<organization|user>/<image>:<version>
```
Comment (Dav-14) [inline components/operator/internal/resources/registries/registries.go:14]: Examples

**Review by paul-nicolas — APPROVED** [review]
LGTM but I don't have all the knowledge about that, maybe wait for gfy review also

File: components/operator/internal/resources/registries/registries.go [hunk 6]
```diff
@@ -9,37 +9,33 @@ import (
...
+	imageOverride, err := settings.GetStringOrEmpty(ctx, stackName, "registries", `"`+registry+`"`, "images", path, "rewrite")
 	if err != nil {
 		return "", err
 	}
 	if imageOverride == "" {
 		imageOverride = path
 	}
 
-	registryEndpoint, err := settings.GetStringOrEmpty(ctx, stackName, "registries", registry, "endpoint")
+	registryEndpoint, err := settings.GetStringOrEmpty(ctx, stackName, "registries", `"`+registry+`"`, "endpoint")
```
Comment (gfyrag) [inline components/operator/internal/resources/registries/registries.go:32]: Why add quotes? You already pass keys values by values, it seems to be useless

File: components/operator/internal/tests/registries_test.go [hunk 7]
```diff
@@ -46,7 +46,7 @@ var _ = Describe("Registries", func() {
 	Context("When overriding image in a settings", func() {
 		var (
 			settings     *v1beta1.Settings
-			registry     = "ghcr.io"
+			registry     = `"ghcr.io"`
```
Comment (gfyrag) [inline components/operator/internal/tests/registries_test.go:49]: still needed?

**Review by gfyrag — APPROVED** [review]

File: components/operator/internal/tests/registries_test.go (hunk 7 above)
Comment (Dav-14) [reply components/operator/internal/tests/registries_test.go:49]: Yes it's to build the settings key

**Pull request merged**